pkg/planner/core/task.go

```diff
@@ -1000,6 +1004,46 @@ func (p *PhysicalTopN) Attach2Task(tasks ...base.Task) base.Task {
 	return attachPlan2Task(p, rootTask)
 }
 
+var annIndexFnNameToMetric = map[string]tipb.VectorDistanceMetric{
+	ast.VecL1Distance:     tipb.VectorDistanceMetric_L1,
+	ast.VecL2Distance:     tipb.VectorDistanceMetric_L2,
+	ast.VecCosineDistance: tipb.VectorDistanceMetric_COSINE,
+
+	// Note: IP is not supported yet. Currently we will throw errors when building IP index.
+	ast.VecNegativeInnerProduct: tipb.VectorDistanceMetric_INNER_PRODUCT,
+}
+
+func fixTopNForANNIndex(p *PhysicalTopN) bool {
```

Comment:
Should set the `IndexID` field in `tipb.ANNQueryInfo`

Reply:
Added.